Error happens during indexing Shielded Expedition chain

The block can be different, one of the first, but code lines are the same all the time
```
2024-02-01T18:27:05.860363Z  INFO namadexer::indexer: Got block 5
    at src/indexer/mod.rs:51
    in namadexer::indexer::get_block with block_height: 5

  2024-02-01T18:27:05.861284Z ERROR namadexer::indexer: Closing block producer task due to an error saving last block: Invalid Transaction data
    at src/indexer/mod.rs:225

Error: InvalidTxData
make: *** [Makefile:49: run_indexer] Error 1
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ path = "src/bin/indexer.rs"
 [dependencies]
 getrandom = { version = "0.2" }
 async-trait = "0.1.51"
-namada_sdk = { git = "https://github.com/anoma/namada", rev = "v0.28.1" }
+namada_sdk = { git = "https://github.com/anoma/namada", rev = "v0.31.0" }
 rand = { version = "0.8", default-features = false }
 rand_core = { version = "0.6", default-features = false }
 tokio = { version = "1.26.0", features = ["rt-multi-thread"] }

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-NAMADA_VERSION := 0.28.1
+NAMADA_VERSION := 0.31.0
 BASE_URL := https://raw.githubusercontent.com/anoma/namada
 URL := $(BASE_URL)/v$(NAMADA_VERSION)/wasm/checksums.json
 

--- a/src/database.rs
+++ b/src/database.rs
@@ -1,24 +1,26 @@
 use crate::queries::insert_block_query;
 use crate::{config::DatabaseConfig, error::Error, utils};
 
-use namada_sdk::core::types::key::common::PublicKey;
+use namada_sdk::types::key::common::PublicKey;
 use namada_sdk::{
+    account::{InitAccount, UpdateAccount},
     borsh::BorshDeserialize,
-    core::types::{
-        address::Address,
-        eth_bridge_pool::PendingTransfer,
-        // key::PublicKey,
-        token,
-        transaction::{
-            account::{InitAccount, UpdateAccount},
-            governance::VoteProposalData,
+    governance::VoteProposalData,
+    tendermint_proto::types::EvidenceList as RawEvidenceList,
+    tx::{
+        data::{
             pgf::UpdateStewardCommission,
             pos::{Bond, Unbond},
             TxType,
         },
+        Tx,
     },
-    proto,
-    tendermint_proto::types::EvidenceList as RawEvidenceList,
+    types::{
+        address::Address,
+        eth_bridge_pool::PendingTransfer,
+        // key::PublicKey,
+        token,
+    },
 };
 use sqlx::postgres::{PgPool, PgPoolOptions, PgRow as Row};
 use sqlx::Row as TRow;
@@ -595,7 +597,7 @@ impl Database {
 
         let mut i: usize = 0;
         for t in txs.iter() {
-            let tx = proto::Tx::try_from(t.as_slice()).map_err(|_| Error::InvalidTxData)?;
+            let tx = Tx::try_from(t.as_slice()).map_err(|_| Error::InvalidTxData)?;
 
             let mut code = Default::default();
             let mut txid_wrapper: Vec<u8> = vec![];
@@ -790,8 +792,7 @@ impl Database {
                         let query = query_builder
                             .push_values(std::iter::once(0), |mut b, _| {
                                 b.push_bind(proposal_id)
-                                    .push_bind(tx_data.vote.is_yay())
-                                    .push_bind(tx_data.vote.is_default_vote())
+                                    .push_bind(tx_data.vote.to_string())
                                     .push_bind(tx_data.voter.encode())
                                     .push_bind(&hash_id);
                             })
@@ -894,8 +895,7 @@ impl Database {
             let mut fee_token: Option<String> = None;
             let mut gas_limit_multiplier: Option<i64> = None;
             if let TxType::Wrapper(txw) = tx.header().tx_type {
-                fee_amount_per_gas_unit =
-                    Some(txw.fee.amount_per_gas_unit.raw_amount().to_string());
+                fee_amount_per_gas_unit = Some(txw.fee.amount_per_gas_unit.to_string_precise());
                 fee_token = Some(txw.fee.token.to_string());
                 let multiplier: u64 = txw.gas_limit.into();
                 // WARNING! converting into i64 might ended up changing the value but there is little

--- a/src/server/tx.rs
+++ b/src/server/tx.rs
@@ -1,25 +1,20 @@
 use crate::error::Error;
-use namada_sdk::core::types::key::common::PublicKey;
-use namada_sdk::core::types::transaction::pos::BecomeValidator;
 use namada_sdk::ibc::apps::transfer::types::msgs::transfer::MsgTransfer;
+use namada_sdk::tx::data::pos::BecomeValidator;
+use namada_sdk::types::key::common::PublicKey;
 use namada_sdk::{
+    account::{InitAccount, UpdateAccount},
     borsh::BorshDeserialize,
-    core::types::{
-        address::Address,
-        eth_bridge_pool::PendingTransfer,
-        transaction::{
-            governance::VoteProposalData,
-            pgf::UpdateStewardCommission,
-            pos::{Bond, Unbond, Withdraw},
-        },
+    governance::VoteProposalData,
+    tx::data::{
+        pgf::UpdateStewardCommission,
+        pos::{Bond, Unbond, Withdraw},
     },
-    core::types::{
-        token,
-        transaction::account::{InitAccount, UpdateAccount},
-    },
+    types::token,
+    types::{address::Address, eth_bridge_pool::PendingTransfer},
 };
 
-use namada_sdk::core::ibc::primitives::proto::Any;
+use namada_sdk::ibc::primitives::proto::Any;
 use prost::Message;
 
 use serde::{Deserialize, Serialize};

--- a/src/tables.rs
+++ b/src/tables.rs
@@ -186,8 +186,7 @@ pub fn get_create_vote_proposal_table(network: &str) -> String {
     format!(
         "CREATE TABLE IF NOT EXISTS {}.vote_proposal (
         vote_proposal_id BYTEA,
-        vote BOOL NOT NULL,
-        vote_default BOOL NOT NULL,
+        vote TEXT NOT NULL,
         voter TEXT NOT NULL,
         tx_id BYTEA NOT NULL
     );",

--- a/src/utils.rs
+++ b/src/utils.rs
@@ -1,4 +1,4 @@
-use namada_sdk::core::types::transaction::TxType;
+use namada_sdk::tx::data::TxType;
 use std::collections::HashMap;
 use std::{env, fs};
 

